Core/src/org/sleuthkit/autopsy/contentviewers/ContactArtifactViewer.java

```diff
@@ -305,16 +396,410 @@ private void updateSection(List<BlackboardAttribute> sectionAttributesList, JLab
+            try {
+                personasMap = super.get();
+
+                if (this.isCancelled()) {
+                    return;
+                }
+                
+                contactUniquePersonasMap.clear();
+                contactUniquePersonasMap.putAll(personasMap);
+                contactUniqueAccountsList.clear();
+                contactUniqueAccountsList.addAll(uniqueAccountsList);
+
+                updatePersonasPanel();
+               
+            } catch (CancellationException ex) {
+                logger.log(Level.INFO, "Persona searching was canceled."); //NON-NLS
+            } catch (InterruptedException ex) {
+                logger.log(Level.INFO, "Persona searching was interrupted."); //NON-NLS
+            } catch (ExecutionException ex) {
+                logger.log(Level.SEVERE, "Fatal error during Persona search.", ex); //NON-NLS
+            }
+        
+        }
+    } 
+    
+    /**
+     * A wrapper class that bags the UI components that need to be updated when 
+     * a persona search task or a create dialog returns.
+     */
+    private class PersonaUIComponents {
+        
+        private final JLabel personaNameLabel;
+        private final JButton personaActionButton;
+    
+        PersonaUIComponents(JLabel personaNameLabel, JButton personaActionButton) {
+            this.personaNameLabel = personaNameLabel;
+            this.personaActionButton = personaActionButton;
+        }
+
+        public JLabel getPersonaNameLabel() {
+            return personaNameLabel;
+        }
+
+        public JButton getPersonaActionButton() {
+            return personaActionButton;
+        }
+    }
+    
+     /**
+     * Action listener for Create persona button.
+     */
+    private class CreatePersonaButtonListener implements ActionListener {
+
+        private final PersonaUIComponents personaUIComponents;
+
+        CreatePersonaButtonListener(PersonaUIComponents personaUIComponents) {
+            this.personaUIComponents = personaUIComponents;
+        }
+         
+        @Override
+        public void actionPerformed(java.awt.event.ActionEvent evt) {
+            // Launch the Persona Create dialog - do not display immediately
+            PersonaDetailsDialog createPersonaDialog = new PersonaDetailsDialog(ContactArtifactViewer.this,
+                    PersonaDetailsMode.CREATE, null, new PersonaCreateCallbackImpl(personaUIComponents), false);
+
+            // Pre populate the persona name and accounts if we have them.
+            PersonaDetailsPanel personaPanel = createPersonaDialog.getDetailsPanel();
+            
+            if (contactName != null) {
+                personaPanel.setPersonaName(contactName);
+            }
+
+            // pass the list of accounts to the dialog
+            for (CentralRepoAccount account : contactUniqueAccountsList) {
+                personaPanel.addAccount(account, "Account found in Contact artifact.", Persona.Confidence.UNKNOWN);
```

Comment:
Should that string be in a @messages block?

Reply:
I'm not sure if this should be in the messages. It's not a message being displayed to the user that needs to be localized - this is going to go in the database.